### DISCUSSION TITLE
No USB Serial Device

### DISCUSSION THREAD

**Discussion by 00-matt** [earlephilhower/arduino-pico#72] [Q&A] [answered]
Hi,

I'm trying to follow the instructions in the README to get it working for the first time but I managed to get stuck on the first step.

After I hold down the BOOTSEL button and plug in the Pico, it shows up a storage device (uf2?) and not a serial device. In the Arduino IDE I've selected the Pico in the board menu, but there are no serial ports to select so uploading a sketch fails.

Any ideas on how I can fix this?

Some possibly relevant dmesg and lsusb output:

```
[ 2200.281435] usb 1-1: new full-speed USB device number 14 using xhci_hcd
[ 2200.430660] usb 1-1: New USB device found, idVendor=2e8a, idProduct=0003, bcdDevice= 1.00
[ 2200.430665] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ 2200.430667] usb 1-1: Product: RP2 Boot
[ 2200.430670] usb 1-1: Manufacturer: Raspberry Pi
[ 2200.430672] usb 1-1: SerialNumber: E0C912D24340
[ 2200.432620] usb-storage 1-1:1.0: USB Mass Storage device detected
[ 2200.432983] scsi host0: usb-storage 1-1:1.0
[ 2201.446254] scsi 0:0:0:0: Direct-Access     RPI      RP2              1    PQ: 0 ANSI: 2
[ 2201.446782] sd 0:0:0:0: Attached scsi generic sg0 type 0
[ 2201.447313] sd 0:0:0:0: [sda] 262144 512-byte logical blocks: (134 MB/128 MiB)
[ 2201.447612] sd 0:0:0:0: [sda] Write Protect is off
[ 2201.447615] sd 0:0:0:0: [sda] Mode Sense: 03 00 00 00
[ 2201.447895] sd 0:0:0:0: [sda] No Caching mode page found
[ 2201.447903] sd 0:0:0:0: [sda] Assuming drive cache: write through
[ 2201.453975]  sda: sda1
[ 2201.455184] sd 0:0:0:0: [sda] Attached SCSI removable disk
```

```
Bus 001 Device 014: ID 2e8a:0003  
```

Thanks.

**Comment by earlephilhower** [comment, marked as answer]
The "upload" just copies a generated .UF2 file to /media/<user>/RPI_RP2/. and the Pico takes care of the rest, so it's a simple process.

Holding down the BOOTSEL makes it appear as a mass storage device, which is how the upload happens, so that's actually fine.

For upload, if you've already entered MSD mode like this, the serial port doesn't matter and you can just hit the upload button.

If for some reason the MSD isn't being mounted, you can manually mount it in the shell/gui and re-try the upload.

**Reply by 00-matt** [reply]
Thank you for the fast reply!

I mounted it manually like this:

```bash
sudo mkdir /media/RPI_RP2
sudo mount -o uid=1000,gid=1000 /dev/sda1 /media/RPI_RP2
```

I could then flash it, and now it shows up as `/dev/ttyACM0` and flashing works normally without having to manually mount anything.
